Do not multiply `AggregateRandom` in `JOIN`s

frouioui opened the pull request:
## Description

This Pull Request fixes the issue described in #11625. When pushing aggregation on a join, we were skipping the multiplication of results for `min` and `max` aggregations, however, we also need to skip the multiplication for `random` aggregations. Random aggregations simply need to get the column form one of the join's sides without multiplying it to the other side of the join.

## Related Issue(s)

- Fixes #11625.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

arthurschreiber reviewed (approved):
❤️

tokikanno commented:
@frouioui : A quick question about `AggregateRandom`, does that mean we'll get random rows from non full group by queries?

BTW: 你會說中文? :-P

frouioui commented:
> A quick question about AggregateRandom, does that mean we'll get random rows from non full group by queries?

@tokikanno, the `AggregateRandom` does not apply a random function to the result, it simply takes the first row that MySQL returns. The results of the `AggregateRandom` will not be "flaky". You can see in the following code snippet how the engine primitive `AggregateRandom` is used:

https://github.com/planetscale/vitess/blob/adeccd2ae1e5b34da0da2fbf89debdc5ccefde2c/go/vt/vtgate/engine/ordered_aggregate.go#L531-L532

> BTW: 你會說中文? :-P

我会一点点 不过我的中文不好 我的大学在北京

frouioui commented:
@systay, you added the `release notes (needs details)` label. I have modified the release notes for all current versions of `v14` and `v15` to include the known issue, and modified `v16`'s release summary to include a note concerning this fix. I'll let you re-review and unlabel the PR if everything looks good.

deepthi commented on doc/releasenotes/16_0_0_summary.md:
```diff
@@ -34,6 +34,13 @@ Other aspects of the VReplication copy-phase logic are preserved:
+
+#### Corrupted results for non-full-group-by queries with JOINs
+
+An issue in versions `<= v14.0.3` and `<= v15.0.0` that generated corrupted results for non-full-group-by queries with a JOIN
```
Why do we need to specify both <= 14.0.3 and <= 15.0.0?

frouioui replied:
To be explicit with where this issue resides. Might help to know that someone running `v15.0.1` looking to upgrade to `v16.0.0` is not affected by that issue and patch.

deepthi replied:
Is it also an issue on 13.0? Or just 14 & 15?

frouioui replied:
It is an issue only in 14 and 15. We were not supporting this kind of query at all before 14.

deepthi reviewed (approved):
👍 on the release notes changes.
